src/utils/AppResponseConverter.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * Italia PagoPA Proxy
+ * Cittadinanza Digitale PagoPA services
+ */
+
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import {
+  ILoginAnonymousResponse,
+  ILoginResponse
+} from "../api/types/LoginResponse";
+import {
+  AckResult,
+  INotificationSubscriptionResponse
+} from "../api/types/NotificationSubscriptionResponse";
+import { ITransactionListResponse } from "../api/types/TransactionResponse";
+import { IWalletResponse } from "../api/types/WalletResponse";
+import { ControllerError } from "../enums/ControllerError";
+import {
+  ILoginAnonymousResponseApp,
+  ILoginResponseApp
+} from "../types/LoginResponseApp";
+import { INotificationSubscriptionResponseApp } from "../types/NotificationSubscriptionResponseApp";
+import {
+  ITransactionApp,
+  ITransactionListResponseApp
+} from "../types/TransactionResponseApp";
+import {
+  IPaymentMethodApp,
+  IWalletResponseApp
+} from "../types/WalletResponseApp";
+
+// Data converter for controllers to translate API responses to Controller responses
+export class AppResponseConverter {
+  public static getLoginFromAPIResponse(
+    loginResponse: ILoginResponse
+  ): ILoginResponseApp | Error {
+    if (loginResponse.apiRequestToken === undefined) {
+      return new Error(ControllerError.ERROR_LOGIN_FAILED);
+    }
+    return {
+      token: loginResponse.apiRequestToken
+    };
+  }
+
+  public static getLoginAnonymusFromAPIResponse(
+    loginAnonymousResponse: ILoginAnonymousResponse
+  ): ILoginAnonymousResponseApp | Error {
+    if (
+      loginAnonymousResponse.apiRequestToken === undefined ||
+      loginAnonymousResponse.apiRequestToken === ""
+    ) {
+      return new Error(ControllerError.ERROR_LOGIN_FAILED);
+    }
+    return {
+      token: loginAnonymousResponse.apiRequestToken as NonEmptyString,
+      type: loginAnonymousResponse.approveTerms.type,
+      title: loginAnonymousResponse.approveTerms.title,
+      privacy: loginAnonymousResponse.approveTerms.properties.privacy,
+      terms: loginAnonymousResponse.approveTerms.properties.terms
+    };
+  }
+
+  public static getWalletFromAPIResponse(
+    walletResponse: IWalletResponse
+  ): IWalletResponseApp | Error {
+    if (walletResponse.data === undefined) {
+      return new Error(ControllerError.ERROR_DATA_NOT_FOUND);
+    }
+    const paymentMethodList: IPaymentMethodApp[] = []; // tslint:disable-line
+    for (const wallet of walletResponse.data) {
```

Comment:
replace the for loop with walletResponse.data.map() then make paymentMethodList immutable